### PR TITLE
:sparkles: Upgrade react-onesignal and refactor OneSignalManager component

### PR DESCRIPTION
**Pull request by issa-diallo**
- Update the react-onesignal package from version 2.0.4 to 3.0.1
    - Integrate state and context management in OneSignalManager
    - Implement createSubscriptionIdOneSignal function to manage subscription IDs through the OneSignal API.
    - Adjust App component structure for better context encapsulation.
    - Add OneSignal type definitions
    - Remove StrictMode from ReactDOM.render to simplify the rendering process.

### CODE REVIEW

File: src/services/oneSignal/OneSignalManager.tsx [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+      const result = await getSubscriptionIdOneSignal(token)
+      if (Array.isArray(result) && result.length > 0) return
+      try {
+        await OneSignal.login(profile.id.toString())
+        const subscriptionId = (window.OneSignal?.User as any).onesignalId
+        assert(subscriptionId, 'Subscription ID is not defined')
+        await createSubscriptionIdOneSignal(token, subscriptionId)
       } catch (error) {
-        console.error('Error initializing OneSignal:', error)
+        console.error(error, 'Error creating subscription ID')
```
Comment (AndreMiras) [inline src/services/oneSignal/OneSignalManager.tsx:36]: Don't we have a component for generic errors too?

File: src/services/api.ts [hunk 2]
```diff
@@ -156,6 +158,21 @@ const updateUser = async (token: string, data: Profile): Promise<Profile> => {
...
 
+const createSubscriptionIdOneSignal = async (token: string, subscriptionId: string): Promise<{}> => {
+  const url = BACKEND_URL + '/onesignal/'
+  const headers = { Authorization: `Token ${token}` }
+  const data = { subscription_id: subscriptionId }
+  const response = await axios.post<OneSignal>(url, data, { headers })
+  return response.data
+}
+
+const getSubscriptionIdOneSignal = async (token: string): Promise<SubscriptionInfo> => {
```
Comment (AndreMiras) [inline src/services/api.ts:169]: `getOneSignalSubscriptionId` feels more natural. Same thing for the above function

File: src/services/oneSignal/OneSignalManager.tsx [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+          safari_web_id: process.env.REACT_APP_SAFARI_WEB_ID as string,
+          allowLocalhostAsSecureOrigin: true,
+          notifyButton: { enable: true },
+          size: 'small'
         })
+        console.log('OneSignal initialized successfully')
+        setOneSignalInitialized(true)
+      }
+      const result = await getSubscriptionIdOneSignal(token)
+      if (Array.isArray(result) && result.length > 0) return
```
Comment (AndreMiras) [inline src/services/oneSignal/OneSignalManager.tsx:29]: Am I missing something or this signature isn't matching with what I'm currently seeing in `SubscriptionInfo`?

**Review by AndreMiras — APPROVED** [review]
We're getting there

File: src/services/oneSignal/OneSignalManager.tsx (hunk 1 above)
Comment (issa-diallo) [reply src/services/oneSignal/OneSignalManager.tsx:36]: I don't want to display the error on the interface

File: src/services/api.ts (hunk 2 above)
Comment (issa-diallo) [reply src/services/api.ts:169]: I Updated

**Review by AndreMiras — APPROVED** [review]

**Pull request merged**